### PR TITLE
feat(tocco-ui): load libphonenumber-js async

### PR DESCRIPTION
**Pull request by khanlifts**
Load libphonenumber-js async to reduce bundle size in initial loading

### CODE REVIEW

File: packages/tocco-ui/src/FormattedValue/typeFormatters/PhoneFormatter.spec.js [hunk 1]
```diff
@@ -8,8 +8,11 @@ describe('tocco-ui', () => {
     describe('typeFormatters', () => {
       describe('PhoneFormatter ', () => {
         test('should format valid phone numbers', () => {
-          const wrapper = shallow(<PhoneFormatter value="+41443886000"/>)
-          expect(wrapper.text()).to.equal('+41 44 388 60 00')
+          const wrapper = mount(<PhoneFormatter value="+41443886000"/>)
+          return wrapper.instance().importLibPhoneNumber().then(() => {
```
Comment (MethodenMann) [inline packages/tocco-ui/src/FormattedValue/typeFormatters/PhoneFormatter.spec.js]: you could introduce a callback that gets invoked as soon as the lib is loaded and use it in the test. i would like that more than invoking a class method like this.

File: packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js [hunk 2]
```diff
@@ -1,75 +1,98 @@
...
+  constructor(props) {
+    super(props)
+    this.state = {libPhoneImport: null, defaultCountry: ''}
+    this.importLibPhoneNumber()
+  }
 
-const removeSpaces = str => str.replace(/\s/g, '')
+  async importLibPhoneNumber() {
+    const libPhoneImport = await import(/* webpackChunkName: "libphonenumber-js" */ 'libphonenumber-js')
+    this.setState({libPhoneImport, defaultCountry: this.DEFAULT_DEFAULT_COUNTRY})
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js]: why do you set state.defaultCountry in this method? There is no cohesion

File: packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js [hunk 3]
```diff
@@ -1,75 +1,98 @@
...
+        const start = caretPosition + offset
+        this.setSelectionRange(start, start)
+      }.bind(inputElement))
+    }
   }
-}
 
-const PhoneEdit = ({value, options, onChange, id, readOnly}) => {
-  let inputElement
+  DEFAULT_DEFAULT_COUNTRY = 'CH'
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js]: const usually are on top

File: packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js [hunk 4]
```diff
@@ -1,75 +1,95 @@
...
-const amountOfSpacesBeforeCaret = (str, caretPosition) => {
-  const spaces = str.substring(0, caretPosition - 1).match(/\s/g)
-  return spaces ? spaces.length : 0
-}
+  amountOfSpacesBeforeCaret = (str, caretPosition) => {
+    const spaces = str.substring(0, caretPosition - 1).match(/\s/g)
+    return spaces ? spaces.length : 0
+  }
+
+  inputElement
```
[the remark this answers is not in the document]
Comment (khanlifts) [reply packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js]: See Line 61.

File: packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js [hunk 5]
```diff
@@ -1,75 +1,102 @@
...
-  const handleChange = e => {
+  handleChange = e => {
+    const defaultCountry = (this.props.options && this.props.options.defaultCountry) || this.state.defaultCountry
     const newValue = e.target.value
-    const parsedNumber = parseNumber(newValue, defaultCountry)
-
-    if (onChange) {
-      const valueNormalized = _isEmpty(parsedNumber) ? removeSpaces(newValue) : formatNumber(parsedNumber, 'E.164')
-      onChange(valueNormalized)
+    const libPhoneImport = this.state.libPhoneImport
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js]: const {libPhoneImport} = this.state

File: packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js [hunk 6]
```diff
@@ -1,75 +1,102 @@
...
+      } else {
+        this.valueNormalized = valueWithoutSpaces
+      }
+      this.props.onChange(this.valueNormalized)
     }
 
     const currentCaretPosition = e.target.selectionStart
-    repositionCaret(newValue, value, currentCaretPosition, inputElement, defaultCountry)
+    this.repositionCaret(newValue, this.props.value, currentCaretPosition)
+    this.setState({...this.state, defaultCountry})
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js]: why do you reset the defaultCountry here?

File: packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js [hunk 7]
```diff
@@ -1,75 +1,101 @@
...
-      </span>}
-    </div>
-  )
+  render() {
+    const displayValue = this.determineDisplayValue()
+
+    return (
+      <div className={this.props.value ? 'input-group' : ''}>
+        <input
+          ref={el => { this.inputElement = el }}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/PhoneEdit.js]: use new react ref api

**Pull request merged**